web/quickfeed_service.go

```diff
@@ -563,17 +563,27 @@ func (s *QuickFeedService) GetOrganization(ctx context.Context, in *connect.Requ
 }
 
 // GetRepositories returns URL strings for repositories of given type for the given course.
-func (s *QuickFeedService) GetRepositories(ctx context.Context, in *connect.Request[qf.URLRequest]) (*connect.Response[qf.Repositories], error) {
+func (s *QuickFeedService) GetRepositories(ctx context.Context, in *connect.Request[qf.CourseRequest]) (*connect.Response[qf.Repositories], error) {
 	course, err := s.db.GetCourse(in.Msg.GetCourseID(), false)
 	if err != nil {
 		s.logger.Errorf("GetRepositories failed: course %d not found: %v", in.Msg.GetCourseID(), err)
 		return nil, connect.NewError(connect.CodeNotFound, errors.New("course not found"))
 	}
 	usrID := userID(ctx)
-	enrol, _ := s.db.GetEnrollmentByCourseAndUser(course.GetID(), usrID)
+	enrol, err := s.db.GetEnrollmentByCourseAndUser(course.GetID(), usrID)
+	if err != nil {
+		s.logger.Error("GetRepositories failed: enrollment for user %d and course %d not found", usrID, course.GetID())
+		return nil, connect.NewError(connect.CodeNotFound, errors.New("enrollment not found"))
+	}
 
 	urls := make(map[string]string)
-	for _, repoType := range in.Msg.GetRepoTypes() {
+	for _, repoType := range repoTypes(enrol) {
+		if repoType == qf.Repository_GROUP && enrol.GroupID == 0 {
```

Comment:
This check should be done in the helper function `repoTypes()`. We should only return a `qf.Repository_GROUP` if the enrollment has a `GroupID > 0`.

Reply:
See code suggestion below.